src/one_dragon_qt/widgets/setting_card/code_editor_setting_card.py

```
@@ -418,36 +451,58 @@ def __init__(self, parent=None, title: str = "代码编辑器", adapter=None):
+            # 插入到最左侧，再加一个弹性空间把它推到左边
+            self.buttonLayout.insertWidget(0, self.external_edit_btn)
+            self.buttonLayout.insertStretch(1)
+
+    def _setup_editor_layout(self) -> None:
+        self.highlighter = PythonHighlighter(self.editor.document())
+        self.viewLayout.addWidget(self.editor)
+
+    def _on_external_edit(self) -> None:
+        import os
+        if self._script_path and os.path.exists(self._script_path):
+            os.startfile(self._script_path)
```

Ruff (0.15.10)

[error] Starting a process without a shell

(S606)